distribution/push_v2.go

```diff
@@ -240,6 +240,11 @@ func (pd *v2PushDescriptor) DiffID() layer.DiffID {
 }
 
 func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.Output) (distribution.Descriptor, error) {
+	if d := pd.layer.ForeignSource(); d != nil {
+		progress.Update(progressOutput, pd.ID(), "Layer already exists")
```

Comment:
I'm wondering if it's better to use different wording indicating that the layer is being skipped before it's a foreign source layer. It could certainly help in debugging to know that this is a not an ordinary layer and it's not expected to exist on the registry after a push.